packages/ra-core/src/form/useWarnWhenUnsavedChanges.tsx

```diff
@@ -28,7 +28,7 @@ const useWarnWhenUnsavedChanges = (
         }
 
         const release = history.block(location => {
-            const isInsideForm = location.pathname.startsWith(
+            const isInsideForm = history.location.pathname.startsWith(
```

Comment:
The problem is that `location.pathname` is pointing to the wrong path, so I use `history.location.pathname`.
I have no clue why this happens